specification/ml/_types/Model.ts

```diff
@@ -44,6 +45,14 @@ export class ModelSnapshot {
   timestamp: long
 }
 
+export class ModelSnapshotUpgrade {
+  job_id: Id
+  snapshot_id: Id
+  state: string
```

Comment:
The state is actually [defined as an enum](https://github.com/elastic/elasticsearch/blob/master/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/job/snapshot/upgrade/SnapshotUpgradeState.java). Should we capture this in the spec?